File: 443. String Compression/solution.java
class Solution {
    // public int Pow(int n ){
    //     int rs =1 ;
    //     while(n>=10){
    //         rs++ ;
    //         n = n/10 ; 
    //     }
    //     return rs  ;
    // }
    public int compress(char[] chars) {
        int current = 0 ;
        int n = chars.length ; 
        int rs  = 0 ;
        for(int i=0; i<n ; i++){
            rs++ ;
            int a = 1 ;
            chars[current]  = chars[i] ;
            current++ ;
            while(i+1 < n && chars[i]==chars[i+1]){    
                i++ ;
                a++ ;      
            }
            
            if(a>1){
                int inter =1 ;
                String rel = String.valueOf(a) ;
                chars[current] = rel.charAt(0) ;
                current++ ;
                while(a>=10){
                    inter++ ;
                    a = a/10 ; 
                    chars[current] = rel.charAt(inter-1) ;
                    current++ ;
                }
                rs = rs+inter ; 
            }
        }
        return rs  ;
    }
}

File: 443. String Compression/solution.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        n = len( chars ) 
        write = 0 
        i = 0 
        while i < n : 
            c = chars[i]
            curr = 1 
            while i+1<n and chars[i+1] == chars[i] : 
                i += 1
                curr += 1 
            chars[write] = c 
            write += 1 
            if curr > 1 :
                for digit in str( curr ) : 
                    chars[write] = digit 
                    write += 1 
            i += 1 
        return write  
